### DISCUSSION TITLE
Walkthrough

### DISCUSSION THREAD

**Discussion by hinogi** [damikun/trouble-training#20] [Q&A]
Is there some tutorial on how this was build step by step? since all at once is a bit overwhelming.

**Comment by damikun** [comment]
> Is there some tutorial on how this was build step by step? since all at once is a bit overwhelming.

Try have a look on Nuke build folder.. nuke run in steps.. 

Or:

For dev, you just need to go to `AP`I, `BFF` and `Idemtity` folders where `*.csproj` file is... Than in each  run 'dotnet watch run'...

This will run 3 services in developer mode manualy... (Equal like you rean any `.net` app

Dont forget, that you need SDK installed like: dotnet, nodejs on your machine.. 

(Note: BFF can take some time since all node_modules must be installed...)

If you have troubles ping hire in discussion.. The repo maybe have issues to build it was time ago... I remember there was some nodejs update that breaks npm install but thats in logs what is wrong usually...
